### PR TITLE
Fix yuzu-cmd missing _NET_WM_ICON on Linux

### PR DESCRIPTION
**Pull request by pineappleEA**
Just like yuzu-qt, currently yuzu-cmd doesn't have a title bar icon and the generic one appears instead. The reason for this is because the icon implantation is a Windows only one using the resource system of Visual Studio with an .rc file. For this pull request a multi-platform approach was taken that abides by the SDL documentation (https://wiki.libsdl.org/SDL_SetWindowIcon). 

Since SDL natively supports only bitmap images a yuzu.bmp image has been added to the dist folder.

If the image was loaded using the SDL_LoadBMP function, for example:
`SDL_Surface* windowIcon = SDL_LoadBMP("yuzu.bmp");`
yuzu-cmd would expect the file in a relative path on the filesystem separate from the binary which isn't very practical.

In order to embed the bitmap inside the binary it's hex code would need to be extracted into a header file but this isn't optimal either since the header has to be modified each time the icon needs to be changed, for example yuzu has two builds with different icons, Mainline and Early Access that both use different icons.

Since Qt isn't used, no resource manager is available so a CMake function was created in order to manage the assets. Instead of creating the header file yuzu_icon.h on src/yuzu_cmd/ CMake creates when it's run and overwrites it when it's run again.

Here's yuzu-cmd before:
image
As you can see it has the generic icon

And yuzu-cmd after:
image
With an icon

While this would probably work on windows too, the resource system is retained just like the yuzu-qt pr so testing is needed

### CODE REVIEW

File: src/yuzu_cmd/emu_window/emu_window_sdl2.cpp [hunk 1]
```diff
@@ -193,6 +194,14 @@ void EmuWindow_SDL2::WaitEvent() {
     }
 }
 
+void EmuWindow_SDL2::SetWindowIcon(){
+ 	SDL_RWops* yuzu_icon_stream = SDL_RWFromMem((void*) yuzu_icon, yuzu_icon_size);
```
Comment (ReinUsesLisp) [inline src/yuzu_cmd/emu_window/emu_window_sdl2.cpp:198]: nit: `SDL_RWFromMem` can potentially fail if the allocation fails, maybe log and abort setting the icon on failure.

File: src/yuzu_cmd/emu_window/emu_window_sdl2.cpp [hunk 2]
```diff
@@ -193,6 +194,14 @@ void EmuWindow_SDL2::WaitEvent() {
...
+    SDL_Surface* windowIcon = SDL_LoadBMP_RW(yuzu_icon_stream, 1);
```
Comment (ReinUsesLisp) [inline src/yuzu_cmd/emu_window/emu_window_sdl2.cpp:199]: nit: Ditto about this potentially failing.

Comment (ReinUsesLisp) [inline src/yuzu_cmd/emu_window/emu_window_sdl2.cpp:199]:
```suggestion
    SDL_Surface* const window_icon = SDL_LoadBMP_RW(yuzu_icon_stream, 1);
```

File: src/yuzu_cmd/emu_window/emu_window_sdl2.cpp (hunk 1 above)
Comment (ReinUsesLisp) [inline src/yuzu_cmd/emu_window/emu_window_sdl2.cpp:198]:
```suggestion
 	SDL_RWops* const yuzu_icon_stream = SDL_RWFromMem((void*) yuzu_icon, yuzu_icon_size);
```

**Review by ReinUsesLisp — COMMENTED** [review]
Some small nits.

As a side note, you can use `ninja clang-format` or `make clang-format` to reformat the whole project. Just make sure not to stage unrelated changes, as clang-format changes the formatting between versions, and we use an old one for validation.

**Comment by bunnei** [comment]
Hi @pineappleEA -- Thank you for your contribution. We've gone ahead and integrated these changes. That said, we do not directly accept pull requests from users who distribute [unofficial builds](https://pineappleea.github.io/) of yuzu, so we are going to close this. In the future, please consider making pull requests under an account that does not distribute unofficial builds.

**Pull request closed without merging**